DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/AllDelegates/AllDelegateItemsViewModel.cs

```diff
@@ -0,0 +1,75 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.AllDelegates
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Models.User;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
+
+    public class AllDelegateItemsViewModel : BaseJavaScriptFilterableViewModel
+    {
+        public readonly IEnumerable<SearchableDelegateViewModel> Delegates;
+
+        public AllDelegateItemsViewModel(
+            int centreId,
+            IEnumerable<DelegateUserCard> delegateUserCards,
+            IEnumerable<(int id, string name)> jobGroups,
+            CentreCustomPromptHelper customPromptHelper
+        )
+        {
+            Delegates = delegateUserCards.Select(
+                delegateUser =>
+                {
+                    var customFields = customPromptHelper.GetCustomFieldViewModelsForCentre(centreId, delegateUser);
+                    var closedCustomPrompts = customPromptHelper.GetClosedCustomPromptsForCentre(centreId);

```

Comment:
I think these closedCustomPrompts should be the same for each delegateUser as it only depends on centreId. Could we define this before the Select statement in order to reduce redundant database calls? Ideally, I think we should be passing closedCustomPrompts in as a parameter rather than defining them inside this constructor.